### ISSUE TITLE
Why the accuracy of classifier on test_seen_feature of CUB is such low?

### ISSUE THREAD

**Issue by Hanzy1996** [akku1506/Feature-Generating-Networks-for-ZSL#1]
First, I run the code of classifier.py on CUB. The accuracy if too low which may lead bad performance on the ZSL task. How can I get better performance on Test Accuracy?

Namespace(attSize=1024, batch_size=64, beta1=0.5, class_embedding='att', dataroot='/home/hanzy/datasets/feature_generating/data', dataset='CUB', gzsl=False, image_embedding='res101', logdir='./logs_classifier/', lr=0.005, manualSeed=9182, matdataset=True, modeldir='./models_classifier/', nepoch=20, preprocessing=False, resSize=2048, standardization=False, test=True, train=True, validation=False)

classification/fc1/kernel:0	
classification/fc1/bias:0	
classification/fc1/kernel/Adam:0	
classification/fc1/kernel/Adam_1:0	
classification/fc1/bias/Adam:0	
classification/fc1/bias/Adam_1:0	
./models_classifier/models_19.ckpt
Model loaded
**### Test Accuracy is:0.08**
classification/fc1/kernel:0	
classification/fc1/bias:0	
classification/fc1/kernel/Adam:0	
classification/fc1/kernel/Adam_1:0	
classification/fc1/bias/Adam:0	
classification/fc1/bias/Adam_1:0	
./models_classifier/models_19.ckpt
Model loaded
Train Accuracy is:0.926666666667
classification/fc1/kernel:0	
classification/fc1/bias:0	
classification/fc1/kernel/Adam:0	
classification/fc1/kernel/Adam_1:0	
classification/fc1/bias/Adam:0	
classification/fc1/bias/Adam_1:0	
./models_classifier/models_19.ckpt
Model loaded
Test Different Labels Accuracy is:0.0

**Comment by Hanzy1996** [comment]
Sorry, I click the fork by mistake. What can I do now?

**Comment by Hanzy1996** [comment]
I change the code as follow in line 167 and the Test Accuracy becomes 66%.
Is it normal or still too low?

acc_per_class[i] = float(np.sum(test_label[idx]==predicted_label[idx])) / float(np.sum(idx))

**Comment by akku1506** [comment]
Hi @Hanzy1996,
The features for CUB dataset are extracted from Resnet-101 model pre-trained on 1000-Imagenet classes. As the model is trained on Imagenet, but features are extracted for CUB dataset, there exists an inherent bias in the features extracted! Thus, performing poorly on CUB as well as SUN dataset.
AWA2 performs well on both train seen and test seen classes as ~30 classes of AWA2 are present in Imagenet dataset.

So, I believe the accuracy that you are obtaining should be fine and should be able to help you reproduce the results of feature generating networks paper on the CUB dataset.  

P.s. thank you for pointing out the minor mistake. I will correct it. Also, test accuraccy on different labels doesn't make sense. I will update it.

**Comment by Pamulapati13** [comment]
Hi @akku1506, 
I was trying to extract the ResNet101 visual features from the images instead of the ones that are provided. I see there is a huge difference in the extracted ones and the given ones. Do you have any idea on the preprocessing steps used to extract the ResNet data?
